go-controller/pkg/node/gateway_localnet.go

```diff
@@ -276,10 +264,10 @@ func (l *localPortWatcher) deleteService(svc *kapi.Service) error {
 				if utilnet.IsIPv6String(externalIP) != isIPv6Service {
 					continue
 				}
-				if _, exists := l.localAddrSet[externalIP]; exists {
+				if _, exists := localAddrSet[externalIP]; exists {
 					iptRules = append(iptRules, getExternalIPTRules(port, externalIP, ip)...)
```

Comment:
We should change this to `getLocalGatewayIPTRules` right? This will allow us to remove the `localAddrSet` calls here

Reply:
>  This will allow us to remove the localAddrSet calls here

you still need to plumb the set, also it will mix things, since `getLocalGatewayIPTRules` returns nodeports rules too.

Let's keep it this way, the code already diverged that is better to keep them as is, instead of creating dependencies between them

Reply:
Agreed Sounds good.